### PR TITLE
Added support for placeholders in replacement text/strings

### PR DESCRIPTION
**Pull request by jshenguru**
Based on discussion here (https://github.com/caddyserver/replace-response/issues/26#issuecomment-2518534936).
Suggested non-disruptive additions to the upstream icholy/replace package.

### CODE REVIEW

File: handler.go [hunk 1]
```diff
@@ -95,14 +97,25 @@ func (h *Handler) Provision(ctx caddy.Context) error {
...
 				if repl.re != nil {
-					tr := replace.RegexpString(repl.re, finalReplace)
+					// This would require the following method in the upstream icholy/replace/replace.go
+					// func RegexpStringWithLateBinding(re *regexp.Regexp, template string, 
+					// 	binding func(string) string) *RegexpTransformer {
+					// 		return RegexpIndexFunc(re, func(src []byte, index []int) []byte {
+					// 			return re.Expand(nil, []byte(binding(template)), src, index)
+					// 	})
+					// }
+					tr := replace.RegexpStringWithLateBinding(repl.re, finalReplace, h.ReplacePlaceholders)
```
Comment (icholy) [inline handler.go]:
```suggestion
					tr := replace.RegexpIndexFunc(re, func(src []byte, index []int) []byte {
						template := h.ReplacePlaceholders(finalReplace)
						return re.Expand(nil, []byte(template), src, index)
					})
```

No upstream changes are required. Just write your code here.

Comment (jshenguru) [reply handler.go]: @icholy Awesome! I fixed the typo (the `re` in your suggested change should be `repl.re`) and tested it. It works like a champ. Now the questions is "How about the conditional branch that deals with non-regexp string?"

File: handler.go [hunk 2]
```diff
@@ -95,14 +97,21 @@ func (h *Handler) Provision(ctx caddy.Context) error {
...
 					tr.MaxMatchSize = 2048
 					transforms[i] = tr
 				} else {
 					finalSearch := placeholderRepl.ReplaceKnown(repl.Search, "")
-					transforms[i] = replace.String(finalSearch, finalReplace)
+					// This would require the following method in the upstream icholy/replace/replace.go
+					// func StringWithLateBinding(old, new string, binding func(string) string) Transformer {
+					// 	return Bytes([]byte(binding(old)), []byte(binding(new)))
+					// }
+					transforms[i] = replace.StringWithLateBinding(finalSearch, finalReplace, h.ReplacePlaceholders)
```
Comment (icholy) [inline handler.go:114]:
```suggestion
					transforms[i] = replace.String(
						h.ReplacePlaceholders(finalSearch),
						h.ReplacePlaceholders(finalReplace),
					)
```

^ this is functionally equivalent to the code you're proposing. Ie, the code you're proposing doesn't actually do late binding.

Comment (jshenguru) [reply handler.go:114]: @icholy You rock! I tested your suggested code and they all work like a champ!
@mholt I closed all previous PRs and created a new one using code that are suggested by @icholy. Please review.
P.S. I've learned a lot about the super dynamic nature of Transformer itself.

Comment (mholt) [reply handler.go:114]: Interesting! Thank you @icholy -- really appreciate that you spared a few minutes of your expertise here.

**Pull request closed without merging**